okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -605,8 +608,8 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
    */
   inner class ReaderRunnable internal constructor(
     internal val reader: Http2Reader
-  ) : Runnable, Http2Reader.Handler {
-    override fun run() {
+  ) : Http2Reader.Handler, () -> Unit {
```

Comment:
oh that’s really neat, I’ve never seen a function type as an implemented interface

Reply:
I'll leave it there, but I was tempted to drop it and also remove Runnable.  But this saves the block allocation I guess.